### PR TITLE
Duplicate performances

### PR DESCRIPTION
**Pull request by wrjren**

### CODE REVIEW

File: src/components/interface/editing/EditDropdown.jsx [hunk 1]
```diff
@@ -11,17 +12,24 @@ import deleteData from '../../../firebase/utils/deleteData';
...
   };
 
+
+  handleDuplicate = async () => {
+    this.props.handleDup();
+    this.handleClose();
+  };
+
   handleClose = () => {
+    console.log('closed')
```
Comment (meganniu) [inline src/components/interface/editing/EditDropdown.jsx]: Clean up console.log

File: src/components/performances/PerformanceDialog.jsx [hunk 2]
```diff
@@ -8,21 +8,36 @@ import PerformanceForm from './PerformanceForm';
...
   handleClickOpen = () => {
     this.setState({ open: true });
   };
 
-  handleClose = () => {
-    this.setState({ open: false });
-  };
-
   render() {
+    console.log('test props', this.props);
```
Comment (meganniu) [inline src/components/performances/PerformanceDialog.jsx]: Clean up console.log

File: src/components/interface/dialog/DialogHeader.jsx [hunk 3]
```diff
@@ -15,13 +15,14 @@ const DialogHeader = ({
   edit,
   onEditClick,
   shouldShowDropdown,
-  title
+  title,
```
Comment (YangStephen) [inline src/components/interface/dialog/DialogHeader.jsx]: Can we alphabetize all of these incoming props

File: src/components/interface/dialog/DialogHeader.jsx [hunk 4]
```diff
@@ -15,13 +15,14 @@ const DialogHeader = ({
...
+  handleDup
 }) => (
   <DialogTitle classes={{ root: classes.dfdialog_title }} disableTypography id="form-dialog-title">
     {title}
     <div style={{ float: 'right', color: 'black' }}>
       {edit && <EditIcon classes={{ root: classes.dfdialog_editIcon }} onClick={onEditClick} />}
-      {shouldShowDropdown && <EditDropDown collectionName={collectionName} docId={docId} />}
+      {shouldShowDropdown && <EditDropDown handleDup={handleDup} collectionName={collectionName} docId={docId} />}
```
Comment (YangStephen) [inline src/components/interface/dialog/DialogHeader.jsx]: lets put handleDup after collectionName and docId

File: src/components/interface/dialog/DialogHeader.jsx [hunk 5]
```diff
@@ -32,6 +33,7 @@ DialogHeader.propTypes = {
   classes: PropTypes.shape().isRequired,
   collectionName: PropTypes.string.isRequired,
   edit: PropTypes.bool,
+  handleDup: PropTypes.func,
```
Comment (YangStephen) [inline src/components/interface/dialog/DialogHeader.jsx]: alphabetize please

File: src/components/interface/dialog/DialogHeader.jsx [hunk 6]
```diff
@@ -42,5 +44,6 @@ DialogHeader.defaultProps = {
   docId: '',
   edit: false,
   onEditClick: null,
+  handleDup: () => {},
```
Comment (YangStephen) [inline src/components/interface/dialog/DialogHeader.jsx]: alphabetize

File: src/components/interface/editing/EditDropdown.jsx [hunk 7]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 
 import Menu from '@material-ui/core/Menu';
+import PropTypes from 'prop-types';
```
Comment (YangStephen) [inline src/components/interface/editing/EditDropdown.jsx]: Let's keep this import above material-ui imports and put it on line 2 with the import React components.

We typically import in groups: React specific, material UI, then components we've made.

File: src/components/interface/editing/EditDropdown.jsx [hunk 8]
```diff
@@ -11,17 +12,24 @@ import deleteData from '../../../firebase/utils/deleteData';
...
     };
   }
 
+
   handleClick = (event) => {
     this.setState({ anchorEl: event.currentTarget });
   };
 
+
+  handleDuplicate = async () => {
```
Comment (YangStephen) [inline src/components/interface/editing/EditDropdown.jsx]: alphabetize these functions.

File: src/components/performances/PerformanceDialog.jsx [hunk 9]
```diff
@@ -8,21 +8,36 @@ import PerformanceForm from './PerformanceForm';
 import DFDialog from '../interface/dialog/DFDialog';
 import DialogHeader from '../interface/dialog/DialogHeader';
 import styles from '../styles';
+import addData from '../../firebase/utils/addData';
 
 class PerformanceDialog extends React.Component {
-  state = {
-    open: false
+  constructor(props) {
+    super(props);
```
Comment (YangStephen) [inline src/components/performances/PerformanceDialog.jsx]: a space between super(props) and initialization of state please!

File: src/components/performances/PerformanceDialog.jsx [hunk 10]
```diff
@@ -8,21 +8,36 @@ import PerformanceForm from './PerformanceForm';
...
 
+
+  handleDup = async () => {
+    let vals = this.props.currentValues;
+    vals.danceTitle = vals.danceTitle + "Duplicate";
+    const { eventId } = this.props;
+    const collectionName = `events/${eventId}/performances`;
+
+    await addData(collectionName, vals);
+    // this.setState({ anchorEl: undefined });
```
Comment (YangStephen) [inline src/components/performances/PerformanceDialog.jsx]: clean up loose comment

File: src/components/performances/PerformanceDialog.jsx [hunk 11]
```diff
@@ -8,21 +8,36 @@ import PerformanceForm from './PerformanceForm';
...
+    this.state = { open: false };
+  }
+
+  handleClose = () => {
+    this.setState({ open: false });
   };
 
+
+  handleDup = async () => {
+    let vals = this.props.currentValues;
```
Comment (YangStephen) [inline src/components/performances/PerformanceDialog.jsx]: this doesn't need to be a const, also we can destructure here.

const {
    currentValues: vals
} = this.props

File: src/components/performances/PerformanceDialog.jsx [hunk 12]
```diff
@@ -8,21 +8,36 @@ import PerformanceForm from './PerformanceForm';
...
+    vals.danceTitle = vals.danceTitle + "Duplicate";
+    const { eventId } = this.props;
```
Comment (YangStephen) [inline src/components/performances/PerformanceDialog.jsx]: Let's merge this and line 25  into one destructure.

File: src/components/performances/PerformanceDialog.jsx [hunk 13]
```diff
@@ -49,6 +64,7 @@ class PerformanceDialog extends React.Component {
         onClick={this.handleClickOpen}
         onClose={this.handleClose}>
         <DialogHeader
+          handleDup={this.handleDup}
```
Comment (YangStephen) [inline src/components/performances/PerformanceDialog.jsx]: alphabetize

File: src/components/performances/PerformanceDialog.jsx [hunk 14]
```diff
@@ -77,13 +93,15 @@ PerformanceDialog.propTypes = {
   }),
   eventId: PropTypes.string.isRequired,
   formType: PropTypes.oneOf(['edit', 'new']),
-  performanceId: PropTypes.string
+  performanceId: PropTypes.string,
+  collectionName: PropTypes.string
```
Comment (YangStephen) [inline src/components/performances/PerformanceDialog.jsx]: alphabetize

File: src/components/performances/PerformanceDialog.jsx [hunk 15]
```diff
@@ -77,13 +93,15 @@ PerformanceDialog.propTypes = {
...
 };
 
 PerformanceDialog.defaultProps = {
   currentValues: {},
   formType: 'edit',
-  performanceId: null
+  performanceId: null,
+  collectionName: ''
```
Comment (YangStephen) [inline src/components/performances/PerformanceDialog.jsx]: alphabetize

File: src/firebase/utils/getData.js [hunk 16]
```diff
@@ -0,0 +1,12 @@
+import db from '../firebase';
+
+const getData = async (collectionName, data) => {
+  db.collection(collectionName).get(data)
+    .then(() => console.log('Document successfully added!'))
+    .catch((error) => {
+      console.error('Error writing document: ', error);
+      return false;
```
Comment (YangStephen) [inline src/firebase/utils/getData.js]: Why is there a return false, theoretically this is an async function so I don't think this return goes back anywhere.

File: src/firebase/utils/getData.js [hunk 17]
```diff
@@ -0,0 +1,12 @@
...
+    });
+};
+
+export default getData;
```
Comment (YangStephen) [inline src/firebase/utils/getData.js]: Can we add a new line at the end of this file.

File: src/components/performances/PerformanceDialog.jsx [hunk 18]
```diff
@@ -8,21 +8,36 @@ import PerformanceForm from './PerformanceForm';
...
+  }
+
+  handleClose = () => {
+    this.setState({ open: false });
   };
 
+
+  handleDup = async () => {
+    let vals = this.props.currentValues;
+    vals.danceTitle = vals.danceTitle + "Duplicate";
```
Comment (YangStephen) [inline src/components/performances/PerformanceDialog.jsx]: Actually coming back now, there is another way we could form this object.

```
const {
    currentValues
} = this.props

const vals = {
    ... currentValues,
    danceTitle: `${currentValues.danceTitle} Duplicate`
}
```

File: src/components/interface/dialog/DialogHeader.jsx (hunk 5 above)
Comment (YangStephen) [reply src/components/interface/dialog/DialogHeader.jsx]: @wrjren This one actually isn't fully alphabetized, can we ensure the entire object is alphabetized. docId should be above edit. Not your fault on this one :)

File: src/components/interface/editing/EditDropdown.jsx [hunk 19]
```diff
@@ -11,17 +12,24 @@ import deleteData from '../../../firebase/utils/deleteData';
 class EditDropdown extends React.Component {
   constructor(props) {
     super(props);
-
     this.state = {
       anchorEl: undefined
     };
   }
 
+
```
Comment (YangStephen) [inline src/components/interface/editing/EditDropdown.jsx]: remove this extra line

File: src/components/interface/editing/EditDropdown.jsx [hunk 20]
```diff
@@ -11,17 +12,24 @@ import deleteData from '../../../firebase/utils/deleteData';
...
   handleClick = (event) => {
     this.setState({ anchorEl: event.currentTarget });
   };
 
+
```
Comment (YangStephen) [inline src/components/interface/editing/EditDropdown.jsx]: another extra line

File: src/components/interface/editing/EditDropdown.jsx (hunk 1 above)
Comment (YangStephen) [reply src/components/interface/editing/EditDropdown.jsx]: This is not resolved

File: src/components/performances/PerformanceDialog.jsx (hunk 10 above)
Comment (YangStephen) [reply src/components/performances/PerformanceDialog.jsx]: This is unresolved.

File: src/components/performances/PerformanceDialog.jsx (hunk 2 above)
Comment (YangStephen) [reply src/components/performances/PerformanceDialog.jsx]: Unresolved.

File: src/components/performances/PerformanceDialog.jsx [hunk 21]
```diff
@@ -75,12 +96,14 @@ PerformanceDialog.propTypes = {
     school: PropTypes.string,
     size: PropTypes.number
   }),
+  collectionName: PropTypes.string,
   eventId: PropTypes.string.isRequired,
   formType: PropTypes.oneOf(['edit', 'new']),
-  performanceId: PropTypes.string
+  performanceId: PropTypes.string,
```
Comment (YangStephen) [inline src/components/performances/PerformanceDialog.jsx]: this comma is not needed.

File: src/firebase/utils/addData.js [hunk 22]
```diff
@@ -1,6 +1,8 @@
 import db from '../firebase';
 
 const addData = async (collectionName, data) => {
+  console.log('add Data collectionName: ', collectionName);
```
[the remark this answers is not in the document]
Comment (YangStephen) [reply src/firebase/utils/addData.js]: unresolved.

File: src/firebase/utils/getData.js (hunk 17 above)
Comment (YangStephen) [reply src/firebase/utils/getData.js]: unresolved.

**Pull request merged**